File: TripService/Models/Destination.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TripService.Models
{
    public class Destination
    {
        public int Id { get; set; }
        public int TripId { get; set; }
        public required string Name { get; set; }
        public required string Location { get; set; }
        public DateTime ArrivalDate { get; set; }
        public DateTime DepartureDate { get; set; }

        [Column("Notes")]
        public string Description { get; set; } = string.Empty;

        public Trip Trip { get; set; } = null!;
    }
}

File: frontend/src/models/Destination.js
export class Destination {
  constructor({ id, tripId, name, location, arrivalDate, departureDate, description }) {
    this.id = id;
    this.tripId = tripId;
    this.name = name;
    this.location = location;
    this.arrivalDate = arrivalDate;
    this.departureDate = departureDate;
    this.description = description;
  }
}
